super_table_pyspark.py

```diff
@@ -0,0 +1,547 @@
+            forecast_24h,
+            id_vars=[
+                "call_timestamp",
+                f"24hr_period_{p}_start",
+                f"24hr_period_{p}_end",
+            ],
+            value_vars=[
+                f"24hr_period_{p}_forecast_west",
+                f"24hr_period_{p}_forecast_east",
+                f"24hr_period_{p}_forecast_central",
+                f"24hr_period_{p}_forecast_south",
+                f"24hr_period_{p}_forecast_north",
+            ],
+        )
+        long_df = long_df.withColumn("split_arr", split(long_df.variable, "_")).select(
+            "*", element_at(col("split_arr"), -1).alias("compass")
+        )
+        long_df = long_df.drop("split_arr", "variable").withColumnRenamed(
+            "value", f"24hr_period_{p}"
+        )
+        long_df_dict[p] = long_df
+
+        # to remove from the main df
+        overlap_cols.extend(
+            [
+                f"24hr_period_{p}_forecast_west",
+                f"24hr_period_{p}_forecast_east",
+                f"24hr_period_{p}_forecast_central",
+                f"24hr_period_{p}_forecast_south",
+                f"24hr_period_{p}_forecast_north",
+                f"24hr_period_{p}_start",
+                f"24hr_period_{p}_end",
+            ]
+        )
+
+    forecast_24h = forecast_24h.drop(*overlap_cols)
+
+    forecast_24h = (
+        forecast_24h.join(long_df_dict[1], how="inner", on="call_timestamp")
+        .join(long_df_dict[2], how="inner", on=["call_timestamp", "compass"])
+        .join(long_df_dict[3], how="inner", on=["call_timestamp", "compass"])
+    )
+
+    return forecast_24h
+
+
+def proc_forecast_2HR(call_timestamp=None):
+    file_path = "forecast-2HR"
+    if call_timestamp:
+        file_path += "_" + formatted_timestamp(call_timestamp)
+    file_path += ".csv"
+
+    schema = schema_forecast_2HR()
+    forecast_2h = get_df_from_csv("forecast-2HR.csv", weather_dir, schema=schema)
+    # Process 2h
+    forecast_2h = forecast_2h.drop("2hr_forecast_area_loc", "2hr_start", "2hr_end")
+    return forecast_2h
+
+
+def proc_bing(call_timestamp=None):
+    file_path = "all-congestion-levels"
+    if call_timestamp:
+        file_path += "_" + formatted_timestamp(call_timestamp)
+    file_path += ".csv"
+
+    bing_key_schema, bing_data_schema = schema_bing()
+
+    bing_data = get_df_from_csv(file_path, bing_dir, schema=bing_data_schema)
+
+    bing_key_table = get_df_from_csv("route_data.csv", bing_dir, schema=bing_key_schema)
+    # Process Bing
+    bing_key_table = bing_key_table.select("camera_id", "direction").withColumnRenamed(
+        "camera_id", "cam_id"
+    )
+    bing_data = bing_data.select(
+        "camera_id", "direction", "call_timestamp", "trafficCongestion"
+    ).withColumnRenamed("camera_id", "cam_id")
+
+    return bing_key_table, bing_data
+
+
+def get_super_table(call_timestamp=None):
+
+    # Camera to locations -
+    # TODO: perhaps run mapping function one time & pull from S3 instead
```

Comment:
Is this mapping attempting to de-normalize the row result for camera?

Reply:
the mapping function tags each camera with the nearest weather station, this function was already called previously and this mapping is saved in the bucket as 'camera_station_mapping.csv', the updated code just pulls this csv and uses it directly.

The schema mismatch errors have been resolved in the latest commits.